src/components/Personalization/index.js

```diff
@@ -43,41 +46,42 @@ const createPersonalization = ({ config, logger, eventManager }) => {
     storeClickMetrics
   } = createClickStorage();
   const getPageLocation = createGetPageLocation({ window });
-  const viewCache = createViewCacheManager();
-  const modules = initDomActionsModules(storeClickMetrics);
-  const executeDecisions = createExecuteDecisions({
+  const viewCache = createViewCacheManager({ createProposition });
+  const modules = initDomActionsModules();
+
+  const preprocess = createPreprocess([remapHeadOffers, remapCustomCodeOffers]);
+
+  const noOpHandler = () => undefined;
+  const domActionHandler = createDomActionHandler({
```

Comment:
I like the idea of action handlers like you added for DOM actions and HTML actions, etc. But it feels like a leaky abstraction since each one has the same modules object passed in (the dom actions modules).  

I'd expect a different modules object for each. For instance...

initDomActionsModules might include actions like customCode, setText, setAttribute, setImageSource, setStyle, move, resize, rearrange, remove, insertAfter, insertBefore

And something like initHTMLActionsModules might include setHtml, replaceHtml, prependHtml, appendHtml (guessing on some of these).

And for in app messages something like initMessagingActionsModules to handle this.

The idea here is to separate each set of actions to a specific schema.

Reply:
This is a good suggestion. I filed a ticket https://jira.corp.adobe.com/browse/PDCL-11018. Let's discuss this next month.